standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java

```diff
@@ -79,13 +78,10 @@ public DataSource create(Configuration hdpConfig, int maxPoolSize) throws SQLExc
     // so that the connection pool can retire the idle connection aggressively,
     // this will make Metastore more scalable especially if there is a leader in the warehouse.
     if ("mutex".equals(poolName)) {
-      int minimumIdle = Integer.valueOf(hdpConfig.get(HIKARI + ".minimumIdle", "2"));
```

Comment:
This change seems unrelated to the purpose of the PR. Moreover, it seems to remove the caching of small integers that happens when `valueOf` is used. Maybe its better to leave things as is.

Reply:
I just seen that we pass the value to a primitive int so calling `valueOf` is completely useless.